packages/backend/src/open_payments/payment_pointer/service.test.ts

```diff
@@ -183,6 +187,37 @@ describe('Open Payments Payment Pointer Service', (): void => {
       ).resolves.toEqual(updatedPaymentPointer)
     })
 
+    test('Deactivating updates expirey dates of existing incoming payments', async (): Promise<void> => {
+      const paymentPointer = await createPaymentPointer(deps)
+
+      const nowMs = new Date('2023-06-01T00:00:00Z').getTime()
+      jest.spyOn(Date, 'now').mockImplementation(() => nowMs)
+      const duration = 30_000
+      const expiresAt = new Date(Date.now() + duration)
+
+      const incomingPayment = await createIncomingPayment(deps, {
+        paymentPointerId: paymentPointer.id,
+        incomingAmount: {
+          value: BigInt(123),
+          assetCode: paymentPointer.asset.code,
+          assetScale: paymentPointer.asset.scale
+        },
+        description: 'Test incoming payment',
+        expiresAt,
+        externalRef: '#123'
+      })
+
+      await paymentPointerService.update({
+        id: paymentPointer.id,
+        status: 'INACTIVE'
+      })
+      const incomingPaymentUpdated = await incomingPayment.$query(knex)
+
+      expect(incomingPaymentUpdated.expiresAt.getTime()).toEqual(
+        expiresAt.getTime() + config.gracePeriodMs - duration
```

Comment:
If you want control over `config.gracePeriodMs` without consequences for other tests, you could use the `withConfigOverride` function.

Reply:
That is good to know, although I don't think that would apply here. I am just reading the grace period here, not changing the value.